src/sentry/releasehealth/metrics.py

```diff
@@ -0,0 +1,138 @@
+    index = indexer.resolve(org_id, UseCase.TAG_VALUE, name)
+    assert index is not None
+    return index
+
+
+def reverse_tag_value(org_id: int, index: int) -> str:
+    str_value = indexer.reverse_resolve(org_id, UseCase.TAG_VALUE, index)
+    assert str_value is not None
+    return str_value
+
+
+class MetricsReleaseHealthBackend(ReleaseHealthBackend):
+    """Gets release health results from the metrics dataset"""
+
+    def get_current_and_previous_crash_free_rates(
+        self,
+        org_id: int,
+        project_ids: Sequence[int],
+        current_start: datetime,
+        current_end: datetime,
+        previous_start: datetime,
+        previous_end: datetime,
+        rollup: int,
+    ) -> ReleaseHealthBackend.CurrentAndPreviousCrashFreeRates:
+        projects_crash_free_rate_dict: ReleaseHealthBackend.CurrentAndPreviousCrashFreeRates = {
+            prj: {"currentCrashFreeRate": None, "previousCrashFreeRate": None}
+            for prj in project_ids
+        }
+
+        previous = self._get_crash_free_rate_data(
+            org_id,
+            project_ids,
+            previous_start,
+            previous_end,
+            rollup,
+        )
+
+        for project_id, project_data in previous.items():
+            projects_crash_free_rate_dict[project_id][
+                "previousCrashFreeRate"
+            ] = self._compute_crash_free_rate(project_data)
+
+        current = self._get_crash_free_rate_data(
+            org_id,
+            project_ids,
+            current_start,
+            current_end,
+            rollup,
+        )
+
+        for project_id, project_data in current.items():
+            projects_crash_free_rate_dict[project_id][
+                "currentCrashFreeRate"
+            ] = self._compute_crash_free_rate(project_data)
+
+        return projects_crash_free_rate_dict
+
+    @staticmethod
+    def _get_crash_free_rate_data(
+        org_id: int,
+        project_ids: Sequence[int],
+        start: datetime,
+        end: datetime,
+        rollup: int,
+    ) -> Dict[int, Dict[str, float]]:
+
+        data: Dict[int, Dict[str, float]] = {}
+
+        session_status = tag_key(org_id, "session.status")
+
+        count_query = Query(
+            dataset=Dataset.Metrics.value,
+            match=Entity("metrics_counters"),
+            select=[Column("value")],
+            where=[
+                Condition(Column("org_id"), Op.EQ, org_id),
+                Condition(Column("project_id"), Op.IN, project_ids),
+                Condition(Column("metric_id"), Op.EQ, metric_id(org_id, "session")),
+                Condition(Column("timestamp"), Op.GTE, start),
+                Condition(Column("timestamp"), Op.LT, end),
+            ],
+            groupby=[
+                Column("project_id"),
+                Column(session_status),
+            ],
+            granularity=Granularity(rollup),
+        )
+
+        # TODO: Add appropriate referrer
```

Comment:
this should be feasible to do now rather than later. just make up any string, like `<servicename>.<methodname>`